connector/src/main/java/org/geysermc/connector/network/translators/bedrock/entity/player/BedrockEmoteTranslator.java

```diff
@@ -52,14 +57,28 @@ public void translate(GeyserSession session, EmotePacket packet) {
             }
         }
 
-        long javaId = session.getPlayerEntity().getEntityId();
-        for (GeyserSession otherSession : session.getConnector().getPlayers()) {
-            if (otherSession != session) {
-                if (otherSession.isClosed()) continue;
-                if (otherSession.getEventLoop().inEventLoop()) {
-                    playEmote(otherSession, javaId, packet.getEmoteId());
-                } else {
-                    session.executeInEventLoop(() -> playEmote(otherSession, javaId, packet.getEmoteId()));
+        if(session.canSendDownstream(PluginMessageUtils.getEMOTE_CHANNEL())){
+            try {
+                ByteArrayOutputStream out = new ByteArrayOutputStream();
+                byte[] emoteId = packet.getEmoteId().getBytes(StandardCharsets.UTF_8);
+                out.write(emoteId.length);
+                out.write(emoteId);
+
+                PluginMessageUtils.sendMessage(session, PluginMessageUtils.getEMOTE_CHANNEL(), out.toByteArray());
+            }catch (Exception e){
+                e.printStackTrace(); //what to do with this
```

Comment:
Should probably log this as a debug message with some form of context like `Failed to send emote data downstream`

Reply:
how can I log? is there a logger?

Reply:
`session.getConnector().getLogger().error("Failed to send emote data downstream!", e);`